demo.cc

```diff
@@ -576,16 +579,134 @@ class Cruncher final {
   void operator=(Cruncher&&) = delete;
 };
 
+class MixpanelUploader final {
+ public:
+  MixpanelUploader(const std::string& demo_id, const std::string& mixpanel_token)
+      : demo_id_(demo_id), mixpanel_token_(mixpanel_token) {}
+
+  inline bool Entry(std::unique_ptr<schema::Base>& entry, size_t index, size_t total) {
+    static_cast<void>(index);
+    static_cast<void>(total);
+
+    struct types {
+      typedef schema::Base base;
+      typedef std::tuple<schema::AnswerRecord> derived_list;
+      typedef bricks::rtti::RuntimeTupleDispatcher<base, derived_list> dispatcher;
+    };
+    types::dispatcher::DispatchCall(*entry, *this);
+
+    return true;
+  }
+
+  inline void Terminate() { std::cerr << '@' << demo_id_ << " MixpanelUploader is done.\n"; }
+
+  inline void operator()(schema::Base&) { /* ignore */
```

Comment:
Interesting, is it required for compilation to not fail? If yes please add a `TODO(dkorolev)`.